content/blog/2020-10-26-october-20-community-gems.md

```diff
@@ -0,0 +1,163 @@
+$ cml-publish movie.mov --md --gitlab-uploads > report.md
+```
+
+Note that we don't currently have an analagous solution for GitHub, because
+GitHub artifacts expire after 90 days (whereas they're permanent in GitLab).
+
+### [Q: I'm getting a mysterious error message, `Failed guessing mime type of file`, when I try to use `cml-publish`. What's going on?](https://discordapp.com/channels/485586884165107732/728693131557732403/763840404675756042)
+
+This error message usually means that the target of `cml-publish`- for example,
+
+```dvc
+$ cml-publish <target file>
+```
+
+is not found. Check for typos in the target filename and ensure that the file
+was in fact generated during the run (if it isn't part of your Git repository).
+We've [opened an issue](https://github.com/iterative/cml/issues/308) to add a
+more informative error message in the future.
+
+### [Q: In my GitHub Actions workflow, I use `dvc metrics diff` to compare metrics generated during the run to `main` and print a table- but the table isn't showing any of the metrics from `main`. What could be happening?](https://discordapp.com/channels/485586884165107732/728693131557732403/768815157034876929)
+
+When a continuous integration runner won't report metrics from previous versions
+of your project (or other branches), that's usually a sign that the runner
+doesn't have access to the full Git history of your project or your metrics
+themselves. Here are a few things to check for:
+
+1. **Did you fetch your Git working tree in the runner?** Functions like
+   `dvc metrics diff` require the Git history to be accessible- make sure that
+   in your workflow, before you run this function, you've done a `git fetch`. We
+   recommend
+
+```dvc
+$ git fetch --prune --unshallow
+```
+
+2. **Are your metrics in your DVC remote?** If your metrics are _cached_ (which
+   they are by default when you create a DVC pipeline), your DVC remote should
+   be accessible to your runner. That means you need to add any credentials as
+   repository secrets (or variables, in GitLab), and do `dvc pull` in your
+   workflow before attempting `dvc metrics diff`.
+
+3. **Are your metrics in your local workspace?** If you are _not_ using a DVC
+   remote, your metric files must be _uncached_ and committed to your Git
+   repository. To explore an example, say you have a pipeline stage that creates
+   `metric.json`:
+
+```dvc
+$ dvc run -n mystage \
+  -m metric.json \
+  train.py
+```
+
+By default, `metric.json` is cached and ignored by Git- which means that if you
+aren't using a DVC remote in your CI workflow, `metric.json` will effectively be
+abandoned on your local machine! You can avoid this by using the `-M` flag
+instead of `-m` in `dvc run`, or manually adding the field `cache: false` to
+your metric in `dvc.yaml`. Be sure to remove your metrics from any `.gitignore`
+files, and commit and push them to your Git repository.
+
+That's all for this month- Happy Halloween! Watch out for scary bugs. 🐛
```

Comment:
can we include some Deevee as a ghost or with a pumpkin 🎃  :) ?

Reply:
definitely